format.swift

```diff
@@ -0,0 +1,503 @@
+#!/usr/bin/env swift
+
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2018 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+
+// MARK: - Logging
+
+// It’s a little unfortunate that we need to duplicate all these functions from
+// swift-syntax-dev-utils. But executing format.swift as a script is significantly
+// faster than running it from a package using `swift run` (~1s vs ~15s), which
+// makes a big difference when running the script multiple times.
+
+func logProcessCommand(executableURL: URL?, arguments: [String]?) {
+  var message = ""
+
+  if let executableURL = executableURL {
+    message += executableURL.path
+  }
+
+  if let arguments = arguments {
+    message += " \(arguments.joined(separator: " "))"
+  }
+
+  print(message)
+}
+
+// MARK: - Paths
+
+enum Paths {
+  static var gitExec: URL {
+    get throws {
+      return try lookupExecutable(for: "git")
+    }
+  }
+
+  static var swiftExec: URL {
+    get throws {
+      return try lookupExecutable(for: "swift")
+    }
+  }
+
+  /// The root of the swift-syntax package
+  static var packageDir: URL {
+    URL(fileURLWithPath: #file)
+      .deletingLastPathComponent()
```

Comment:
Some places we use `return` in `var`s like above or a bit below. 

Do we have a specific style guide for that in SwiftSyntax. Just to keep consistency?

Reply:
I've been using [`SwiftFormat`](https://github.com/nicklockwood/SwiftFormat) for years and it has the ability to keep/add a `return` statement in such cases or remove (based on config file). I'm new to `swift-format` so I'm unsure if this tool has similar abilities - but it would be really cool to have something like it here. 

Oh, and as I was sneaking through the code, my nitpick-dar (that's nitpick radar) also did a little beep-beep about this `return`s. 📡😂

Reply:
We don’t have a consistent style here. I personally dislike the implicit return, especially if the returned expression spans multiple lines (which is very subjective and not based on facts) but acknowledge that the general style seems to be to omit them. I have removed the `return`s.

AFAIK there’s no rule for that in swift-format. It would be nice to add one though.

Reply:
Opened an issue here: https://github.com/apple/swift-format/issues/582